### ISSUE TITLE
API Router add helper/fixture

### ISSUE THREAD

**Issue by aviramha** [tiangolo/fastapi#753]
### Is your feature request related to a problem

Right now we have dependencies injection for API router, which can easily wrap API calls and run some logic before passing it to the method.
Besides that, if we want to access any value generated by the dependency we need to declare it for each path/api call. 

### The solution you would like

Add optional keyword argument to the Depends class, called name.
In regular usage of Depends it ignores it, but if used on APIRoute it delivers the result of that dependency as that keyword argument to the function.
for example:

```
async def get_user() -> User:
    return user

app.include_router(
    dependencies=[Depends(get_user, name=user)],
)

def method_under_router(user: User):
 
```
I would like feedback before implementing it myself. I got this idea from @jklang on gitter and it sounds like a good feature to have.

**Comment by tiangolo** [comment]
I would prefer not to have something like that as it would tie each value to the type returned by the dependency while it's already tied to the dependable object (the function), so it would add complexity and cognitive burden, to understand how it works in some situations but it's different in others. It would also look like a request body.

Nevertheless, you can do:

```Python
async def get_user() -> User:
    return user

app.include_router(
    dependencies=[Depends(get_user)],
)

def method_under_router(user: User = Depends(get_user)):
```

and it would do the same. It would not call the dependency twice, as there's a dependency cache in place (that you can disable). With that you exchange the proposed `name="user"` (I imagine that's what you meant) with a `= Depends(get_user)`. It's just a bit more characters, but it's way more explicit and obvious. And the implementation is already there.

**Comment by dmontagu** [comment]
I agree with @tiangolo, I think this is likely to quickly become an unmaintainable source of confusion if used heavily in a larger codebase.

I personally use [this cbv pattern](https://fastapi-utils.davidmontague.xyz/user-guide/class-based-views/) to reduce the need to repeat dependencies, though obviously it's a very different change than proposed here.

**Comment by aviramha** [comment]
Thanks for the comments - I'll close the issue.
@dmontagu Great to see your fastapi_utils repo - it's really useful! I'll use it!
